inst/tests/tests.Rraw

```diff
@@ -15653,7 +15653,7 @@ test(2071.09, any_na(data.table(expression(1))), error="Unsupported column type
 test(2071.10, dcast(data.table(a=1, b=1, l=list(list(1))), a ~ b, value.var='l'),
               data.table(a=1, `1`=list(list(1)), key='a'))
 test(2071.11, dcast(data.table(a = 1, b = 2, c = 3), a ~ b, value.var = 'c', fill = '2'),
-              data.table(a=1, `2`=3, key='a'))
+              data.table(a=1, `2`=3, key='a'), warning="Coercing 'character' RHS to 'double'")
```

Comment:
I believe this warning is consistent with alerting user to string->double coercion elsewhere, but could be considered a breaking change, WDYT?

Reply:
consistent with what I wrote in #5549, I think this warning is confusing, because fill is not actually used in the output. I would like to merge that other PR, and if we do, then this warning will go away (and I guess we will have to modify this test).

If you want to test the coercion warning, I would suggest a more complicated case when the fill is actually used in the output, such as
```
> dcast(data.table(a = c(1,1,2), b = c(1,2,1), c = 3), a ~ b, value.var = 'c', fill='2')
Key: <a>
       a     1     2
   <num> <num> <num>
1:     1     3     3
2:     2     3     2
```
Does that make sense?

Reply:
Makes perfect sense. IINM these tests are just about coverage, so your modified example will be better for now. Then hopefully this warning will go away after #5549 -- in fact we should make sure to include this as a test case there.

Reply:
lgtm